### PR TITLE
bib: switch to use bootc install to-filesystem (HMS-3453)

### PR DESCRIPTION
**Pull request by mvo5**
[draft as this requires https://github.com/osbuild/images/pull/571 first]

This replaces https://github.com/osbuild/bootc-image-builder/pull/304 which did not implement customizations.
With this approach we keep the user customizations we had before (it does drop groups for now mostly because that seems less used and more risky).

### CODE REVIEW

File: bib/cmd/bootc-image-builder/image.go [hunk 1]
```diff
@@ -77,9 +77,9 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
 	}
 
 	img := image.NewBootcDiskImage(containerSource)
+	// XXX: hardcoded, get from the image instead
+	img.SElinux = "targeted"
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply bib/cmd/bootc-image-builder/image.go:81]: img.SELinux is the way I would spell it.

[the remark this answers is not in the document]
Comment (mvo5) [reply bib/cmd/bootc-image-builder/image.go]: I followed the existing customization in `github.com/osbuild/images/pkg/manifest/os.go` here but I agree that it's weird and it's fixed in the https://github.com/osbuild/images/pull/571 PR now too (thanks to Ondrej for also pointing it out there).

File: bib/cmd/bootc-image-builder/image.go [hunk 2]
```diff
@@ -78,7 +78,8 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
 
 	img := image.NewBootcDiskImage(containerSource)
 	img.Users = users.UsersFromBP(customizations.GetUsers())
-	img.Groups = users.GroupsFromBP(customizations.GetGroups())
```
Comment (mvo5) [inline bib/cmd/bootc-image-builder/image.go:81]: I will add this back once https://github.com/osbuild/images/pull/593 is merged

File: bib/cmd/bootc-image-builder/image.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvo5) [reply bib/cmd/bootc-image-builder/image.go]: This is fixed now

File: bib/cmd/bootc-image-builder/image.go (hunk 2 above)
Comment (achilleas-k) [reply bib/cmd/bootc-image-builder/image.go:81]: The PR is in the queue and will probably be merged in a few mins but we're also going to need a release and dependency update so I think it's okay to merge this and open a follow-up immediately after.

**Review by achilleas-k — APPROVED** [review]

**Comment by mvo5** [comment]
Hrm, hrm, this is ready now (including groups) *except* the cross arch building needs either:
- https://github.com/containers/bootc/pull/462 (released in the relevant places)
or
- https://www.mail-archive.com/qemu-devel@nongnu.org/msg1034508.html (released in the relevant places)

File: bib/cmd/bootc-image-builder/image.go (hunk 2 above)
Comment (mvo5) [reply bib/cmd/bootc-image-builder/image.go:81]: Groups are back now too, I close this conversation.

**Review by cgwalters — COMMENTED** [review]
We need to bind mount in the bootc container root (target root), not the bib root right?

**Comment by mvo5** [comment]
> We need to bind mount in the bootc container root (target root), not the bib root right?

Indeed, sorry that I overlooked this - I reverted this and opened https://github.com/osbuild/images/pull/604 which allows us to make bib a frankenstein to build cross-arch. The bind mount idea is also (briefly) discussed there.

**Comment by mvo5** [comment]
The tests are green now, we should probably still wait until https://github.com/osbuild/bootc-image-builder/pull/357 is landed to ensure we do not regress there (I think we will need https://github.com/osbuild/osbuild/pull/1740 and https://github.com/osbuild/images/pull/605). But getting really close - and so is the deadline :(

**Review by ondrejbudai — DISMISSED** [review]
I love this! No concerns, let's get this in!

**Comment by mvo5** [comment]
I updated this now to include https://github.com/osbuild/bootc-image-builder/pull/360 and also the latest update in images which brings in https://github.com/osbuild/images/pull/605 - with that and the "legacy" bootc workaround for the cross arch build this should finally be ready.

**Review by achilleas-k — DISMISSED** [review]
LGTM.  Very exciting!!

... but the linter is unhappy.

**Comment by achilleas-k** [comment]
Hm, the linter is being an idiot again.  Let me dig into this a bit.

**Comment by achilleas-k** [comment]
Ok, I think I fixed it.  Probably.

**Review by achilleas-k — APPROVED** [review]

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**